src/App.tsx

```diff
@@ -8,55 +8,134 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { getUsers } from './utils/helpers';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
+import { LoadingItems } from './types/LoadingItems';
+import { HasErrorItem } from './types/ErrorMessage';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[] | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [comments, setComments] = useState<Comment[] >([]);
+  const [isLoading, setIsLoading] = useState<LoadingItems>('');
+  const [hasError, setHasError] = useState<HasErrorItem>('');
+
+  useEffect(() => {
+    const fetch = async () => {
```

Comment:
```suggestion
    const fetchUsers = async () => {
```

let's user more informative naming